Extension/src/Utility/Sandbox/sandbox.ts

```diff
@@ -0,0 +1,111 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Context, createContext, runInContext, Script } from 'vm';
+import { stringify } from '../System/json';
+import { verbose } from '../Text/streams';
+import { CreateOptions, ScriptError } from './interfaces';
+
+/**
+ * Creates a reusable safe-eval sandbox to execute code in.
+ */
+export function createSandbox(): <T>(code: string, context?: any) => T {
+    const sandbox = createContext({});
+    return (code: string, context?: any) => {
+        const response = `SAFE_EVAL_${Math.floor(Math.random() * 1000000)}`;
+        sandbox[response] = {};
+        if (context) {
+            Object.keys(context).forEach((key) => (sandbox[key] = context[key]));
+            runInContext(
+                `try {  ${response} = ${code} } catch (e) { ${response} = undefined }`,
+                sandbox
+            );
+            for (const key of Object.keys(context)) {
+                delete sandbox[key];
+            }
+        } else {
+            try {
+                runInContext(`${response} = ${code}`, sandbox);
+            } catch (e) {
+                sandbox[response] = undefined;
+            }
+        }
+        return sandbox[response];
+    };
+}
+
+export const safeEval = createSandbox();
+
+/**
+ * A class that provides the ability to execute code from the user in a safe way.
+ * (it does so using node's VM support.)
+ */
+export class Sandbox {
+    context: Context;
+
+    constructor(initializeContext: Record<string, any> = {}) {
+        this.context = createContext({
+            exports: {},
+            ...initializeContext,
+            console: {
+                log: console.log,
+                error: console.error,
+                debug: console.debug,
+                info: console.info,
+                warn: console.warn,
+                verbose: verbose
+            },
+            JSON: {
+                stringify: (obj: any) => stringify(obj),
+                parse: (str: string) => JSON.parse(str)
+            }
+
+        });
+    }
+
+    protected require(module: string) {
+        return require(module);
+    }
+
+    /**
+     * Creates an adhoc function from raw JS code.

```

Comment:
Should the JS and TS references be replaced by JavaScript and TypeScript or is this supposed to refer to something different?

Reply:
I'm removing the comments, they refer to code that was removed.

Reply:
Corrected.